wgpu-remote/src/server.rs

```diff
@@ -28,10 +28,23 @@ pub extern "C" fn wgpu_server_instance_request_adapter(
     ids: *const wgn::AdapterId,
     id_length: usize,
 ) -> i8 {
+    extern "C" fn request_adapter_callback(
```

Comment:
I assumed we'd want the server to block until the callback fires, but we could also make this async if we wanted